### PR TITLE
[Merged by Bors] - Expand merge migration docs

### LINKED ISSUE #3424: Lighthouse book - Merge Migration section confusing
## Description

I found the [Merge Migration](https://lighthouse-book.sigmaprime.io/merge-migration.html) instructions confusing. Mainly because I wasn't sure if new options were needed right now or should be added closer to when the merge is supposed to occur.

## Version

v2.5.1

## Present Behaviour

The confusion started here:

---
As of [v2.4.0](https://github.com/sigp/lighthouse/releases/tag/v2.4.0) Lighthouse can be considered to have two modes:

  * **Pre-merge**: --execution-endpoint flag is not provided.
  * **Post-merge**: --execution-endpoint flag is provided.

 A pre-merge node, by definition, will fail to transition through the merge. Such a node must be upgraded before the Bellatrix upgrade.

---

I think the instructions could be more explicit, I found these [CoinCashew instructions informative](https://www.coincashew.com/coins/overview-eth/ethereum-merge-upgrade-checklist-for-home-stakers-and-validators). Actually, there should be a `Lighthouse Flags & Options` section that would include "required" and "typical" options that are needed to run a bn and vc.

## Expected Behaviour

I think this is more correct and to the point:

---
As of [v2.4.0](https://github.com/sigp/lighthouse/releases/tag/v2.4.0) Lighthouse bn and vc new options should be added ASAP but is required before the Bellatrix upgrade. And for testnets they Goerli/Prater this need to be done ASAP.

* For beacon node add `--execution-endpoint` and `--execution-jwt` options
* For validator clients add [`--suggested-fee-recipient`](https://lighthouse-book.sigmaprime.io/suggested-fee-recipient.html)

If these flags are not present your system will fail to transition through the merge.

---

### PR DESCRIPTION
**Pull request by michaelsproul**
## Issue Addressed

Resolves #3424

## Proposed Changes

This PR expands the merge migration docs to include (hopefully) clearer guidance on the steps required. It's inspired by @winksaville's work in #3426 but takes a more drastic approach to rewriting large sections.

* Add a prominent _When?_ section
* Add links to execution engine configuration guides
* Add links to community guides
* Fix the location of the _Strict fee recipient_ section

### CODE REVIEW

File: book/src/merge-migration.md [hunk 1]
```diff
@@ -1,20 +1,69 @@
...
+## Necessary Configuration
 
-- "Pre-merge": `--execution-endpoint` flag *is not* provided.
-- "Post-merge": `--execution-endpoint` flag *is* provided.
+There are two configuration changes required for a Lighthouse node to operate correctly throughout
+the merge:
 
-A "pre-merge" node, by definition, will fail to transition through the merge. Such a node *must* be
-upgraded before the Bellatrix upgrade.
+1. You *must* run a local execution engine such as Geth or Nethermind alongside Lighthouse.
```
Comment (winksaville) [inline book/src/merge-migration.md]: Does local mean same machine or same local network?

File: book/src/merge-migration.md [hunk 2]
```diff
@@ -1,20 +1,69 @@
...
+## When?
+
+You must configure your node to be merge-ready before the Bellatrix fork occurs on the network
+on which your node is operating.
+
+* **Mainnet**: the Bellatrix fork epoch has not yet been announced. It's possible to set up a
+  merge-ready node now, but some execution engines will require additional configuration. Please see
+  the secton on [Execution engine configuration](#execution-engine-configuration) below.
+
+* **Goerli (Prater)**, **Ropsten**, **Sepolia**, **Kiln**: you must have a merge-ready configuration
```
Comment (winksaville) [inline book/src/merge-migration.md:32]: Are these all ready "post-merge"? I know goerli/prater is :)

File: book/src/merge-migration.md [hunk 3]
```diff
@@ -1,20 +1,69 @@
...
+  right now.
+
+## Connecting to an execution engine
+
+The Lighthouse beacon node must connect to an execution engine in order to validate the transactions
+present in post-merge blocks. Two new flags are used to configure this connection:
+
+- `--execution-endpoint <URL>`: the URL of the execution engine API. Often this will be
+  `http://localhost:8551`.
+- `--execution-jwt <FILE>`: the path to the file containing the JWT secret shared by Lighthouse and the
```
Comment (winksaville) [inline book/src/merge-migration.md:42]: Provide the `openssh rand ...` command as an example of how to create to create the jwt secret file.

**Review by winksaville — COMMENTED** [review]
Nice!

One other suggestion it would be nice if there was a page that documented all of the flags and options for bn & vc's.

File: book/src/merge-migration.md (hunk 2 above)
Comment (michaelsproul) [reply book/src/merge-migration.md:32]: Yep! Goerli is actually only post-Bellatrix, but hasn't merged with the execution chain. The rest have completed the merge

File: book/src/merge-migration.md (hunk 1 above)
Comment (michaelsproul) [reply book/src/merge-migration.md]: Good point. I'll clarify local to "user controlled"

File: book/src/merge-migration.md (hunk 3 above)
Comment (michaelsproul) [reply book/src/merge-migration.md:42]: I think the UX for asking users to run `openssl` and set file permissions manually is pretty poor, so I'd prefer to just refer to the execution node guides which give the flags for auto-generating a JWT secret (most ELs will generate the secret at the path given if it doesn't exist)

**Review by winksaville — APPROVED** [review]
LGTM!

**Comment by michaelsproul** [comment]
Thanks Wink!

File: book/src/merge-migration.md [hunk 4]
```diff
@@ -1,20 +1,69 @@
...
+the merge. Merge releases are available now for all testnets.
+
+## When?
+
+You must configure your node to be merge-ready before the Bellatrix fork occurs on the network
+on which your node is operating.
+
+* **Mainnet**: the Bellatrix fork epoch has not yet been announced. It's possible to set up a
+  merge-ready node now, but some execution engines will require additional configuration. Please see
+  the secton on [Execution engine configuration](#execution-engine-configuration) below.
```
Comment (macladson) [inline book/src/merge-migration.md]:
```suggestion
  the section on [Execution engine configuration](#execution-engine-configuration) below.
```

File: book/src/merge-migration.md [hunk 5]
```diff
@@ -1,20 +1,69 @@
...
+
+Each execution engine has its own flags for configuring the engine API and JWT. Please consult
+the relevant page for your execution engine for the required flags:
+
+- [Geth: Connecting to Consensus Clients](https://geth.ethereum.org/docs/interface/consensus-clients)
+- [Nethermind: Running Nethermind Post Merge](https://docs.nethermind.io/nethermind/first-steps-with-nethermind/running-nethermind-post-merge)
+- [Besu: Prepare For The Merge](https://besu.hyperledger.org/en/stable/HowTo/Upgrade/Prepare-for-The-Merge/)
+
+Once you have configured your execution engine to open up the engine API (usually on port 8551) you
+should add the URL to your `lighthouse bn` flags with `--execution-endpoint URL`, as well as
```
Comment (macladson) [inline book/src/merge-migration.md]:
```suggestion
should add the URL to your `lighthouse bn` flags with `--execution-endpoint <URL>`, as well as
```

File: book/src/merge-migration.md [hunk 6]
```diff
@@ -1,20 +1,69 @@
...
+the path to the JWT secret with `--execution-jwt`.
```
Comment (macladson) [inline book/src/merge-migration.md]:
```suggestion
the path to the JWT secret with `--execution-jwt <FILE>`.
```

File: book/src/suggested-fee-recipient.md [hunk 7]
```diff
@@ -168,4 +163,21 @@ curl -X DELETE \
...
+local execution engine for payload construction, where a strict fee recipient check will still be applied.
+
+## FAQ
+
+### Why do I have to nominate an Ethereum address as the fee recipient?
+
+You might wonder why the validator can't just accumulate transactions fees in the same way that it
+accumulates other staking rewards. The reason for this is that transaction fees are computed and
+validated by the execution node, and therefore need to be paid to an address that exists on the
+execution chain. In the initial merge fork (Bellatrix) validators have no presence
```
Comment (macladson) [inline book/src/suggested-fee-recipient.md]: Looks like this sentence got cut short?

**Review by macladson — CHANGES_REQUESTED** [review]

File: book/src/suggested-fee-recipient.md (hunk 7 above)
Comment (michaelsproul) [reply book/src/suggested-fee-recipient.md]: oh yeah good catch, I must have got distracted :P

Comment (michaelsproul) [reply book/src/suggested-fee-recipient.md]: updated now, thanks!

**Review by macladson — APPROVED** [review]
Looks good!

**Pull request closed without merging**